plugin/evm/atomic/state/interfaces/interfaces.go

```diff
@@ -0,0 +1,155 @@
+package interfaces
```

Comment:
Ideally have these in the package where they are needed. Creating an interfaces package is really strange imo.
Also, having them in a shared interfaces package means it will be harder to have narrow interfaces per package (i.e. package A might only need method X, package B needs methods X and Y, but if they both import interfaces.Interface they would both require X and Y - not great)
I usually have an `interfaces.go` file in the package where the interfaces are needed, since sometimes it's hard to know where to define an interface if it's used in different files in that package.

Reply:
I agree with this and would like to see less "shared interfaces" between packages, even though the existing code doesn't adhere to this.
This goes hand in hand with returning concrete types instead of interfaces, and paying more attention to which methods/fields are exported (as they are "the interface".